consensus/src/sync/block_queue.rs

```diff
@@ -432,6 +433,18 @@ impl<N: Network> Stream for Inner<N> {
                         )));
                     }
                 }
+
+                // TODO this case should be handled
+                PushOpResult::Head(Err(result), hash) => {}
+
+                PushOpResult::Buffered(Err(result), hash) => {
+                    self.pending_blocks.remove(&hash);
```

Comment:
I think this was a bug in the code, because, previously, the error cases were not handled, so if a block was rejected by the blockchain it remained inside `pending_blocks`and nobody cleared that invalid block